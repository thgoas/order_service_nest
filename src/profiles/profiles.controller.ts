import {
  Controller,
  Get,
  Post,
  Body,
  Patch,
  Param,
  Delete,
  HttpException,
  HttpStatus,
} from '@nestjs/common';
import { ProfilesService } from './profiles.service';
import { Prisma } from '@prisma/client';

@Controller('profiles')
export class ProfilesController {
  constructor(private readonly profilesService: ProfilesService) {}

  @Post()
  create(@Body() createProfileDto: Prisma.ProfileCreateInput) {
    if (createProfileDto.name === '' || createProfileDto.name === null) {
      throw new HttpException('cannot be empty!', HttpStatus.FORBIDDEN);
    }
    if (
      createProfileDto.description === '' ||
      createProfileDto.description === null
    ) {
      throw new HttpException('cannot be empty!', HttpStatus.FORBIDDEN);
    }

    return this.profilesService.create(createProfileDto);
  }

  @Get()
  findAll() {
    return this.profilesService.findAll();
  }

  @Get(':id')
  findOne(@Param('id') id: string) {
    return this.profilesService.findOne(id);
  }

  @Patch(':id')
  update(
    @Param('id') id: string,
    @Body() updateProfileDto: Prisma.ProfileUpdateInput,
  ) {
    return this.profilesService.update(id, updateProfileDto);
  }

  @Delete(':id')
  remove(@Param('id') id: string) {
    return this.profilesService.remove(+id);
  }
}
