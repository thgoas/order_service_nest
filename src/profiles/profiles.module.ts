import { Module } from '@nestjs/common'
import { ProfilesController } from './profiles.controller'
import { ProfilesService } from './profiles.service'
import { CommonModule } from 'src/common/common.module'

@Module({
  imports: [CommonModule],
  controllers: [ProfilesController],
  providers: [ProfilesService],
})
export class ProfilesModule {}
