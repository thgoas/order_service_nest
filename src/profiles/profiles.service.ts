import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
import { Prisma } from '@prisma/client';
import { PrismaService } from '../common/prisma/prisma.service';

@Injectable()
export class ProfilesService {
  constructor(private readonly prisma: PrismaService) {}
  async create(createProfileDto: Prisma.ProfileCreateInput) {
    try {
      return await this.prisma.profile.create({
        data: {
          ...createProfileDto,
        },
      });
    } catch (error) {
      console.log(error.code);
      if ((error.code = 'P2002')) {
        throw new HttpException(
          `name ${createProfileDto.name} already exists in the database!`,
          HttpStatus.UNAUTHORIZED,
        );
      }
    }
  }

  findAll() {
    try {
      return this.prisma.profile.findMany();
    } catch (error) {
      console.error(error);
    }
  }

  async findOne(id: string) {
    try {
      return await this.prisma.profile.findUnique({
        where: { id },
      });
    } catch (error) {
      console.error(error);
    }
  }

  async update(id: string, updateProfileDto: Prisma.ProfileUpdateInput) {
    const date = new Date();
    try {
      return await this.prisma.profile.update({
        where: { id },
        data: {
          ...updateProfileDto,
          updated_at: date,
        },
      });
    } catch (error) {
      console.error('Update Error', error);
      throw new HttpException(error.meta.cause, HttpStatus.FOUND);
    }
  }

  remove(id: number) {
    return `This action removes a #${id} profile`;
  }
}
