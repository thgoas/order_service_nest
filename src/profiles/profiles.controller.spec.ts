import { Test, TestingModule } from '@nestjs/testing';
import { ProfilesController } from './profiles.controller';
import { ProfilesService } from './profiles.service';
import { CommonModule } from '../common/common.module';
import { HttpException, HttpStatus } from '@nestjs/common';

describe('ProfilesController', () => {
  let controller: ProfilesController;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      imports: [CommonModule],
      controllers: [ProfilesController],
      providers: [ProfilesService],
    }).compile();

    controller = module.get<ProfilesController>(ProfilesController);
  });

  it('should be defined', () => {
    expect(controller).toBeDefined();
  });

  it('should throw HttpException for empty string', () => {
    const data = {
      description: '',
      name: '',
    };
    expect(() => controller.create(data)).toThrow(
      new HttpException('cannot be empty!', HttpStatus.FORBIDDEN),
    );
  });
  it('should not throw HttpException for non-empty string', () => {
    const data = {
      description: 'exist',
      name: 'exist',
    };
    expect(() => controller.create(data)).not.toThrow();
  });
});
