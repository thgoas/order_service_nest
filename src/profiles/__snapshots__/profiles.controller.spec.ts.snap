// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilesController should throw an http exception: cannot be empt! 1`] = `"cannot be empty!"`;

exports[`ProfilesController should throw an http exception: cannot be empty! 1`] = `"cannot be empty!"`;
